### DISCUSSION TITLE
how to group `cftime` coordinate by `week`?

### DISCUSSION THREAD

**Discussion by aaronspring** [pydata/xarray#6375] [Q&A] [answered]
Suppose you have to work with `cftime` coordinates, how do you group by `week`?
The only way I found working was converting to `datetime` calendar with `convert_calendar`.

Is there a `cftime` solution? Is there a reason why `CFTimeIndex` doesnt implement `week`?

```python
import xarray as xr
xr.__version__ # '2022.3.0'

import cftime
cftime.__version__ # '1.6.0'

ds = xr.tutorial.load_dataset('air_temperature', use_cftime=True)

ds.convert_calendar(calendar='standard').groupby('time.week').mean() # works

ds.groupby('time.week').mean() # AttributeError: 'CFTimeIndex' object has no attribute 'isocalendar'

ds.groupby(ds.time.dt.isocalendar().week).mean() # AttributeError: 'CFTimeIndex' object has no attribute 'isocalendar'


ds = xr.tutorial.load_dataset('air_temperature', use_cftime=False)

ds.groupby('time.week').mean() # FutureWarning: dt.weekofyear and dt.week have been deprecated. Please use dt.isocalendar().week instead.

ds.groupby(ds.time.dt.isocalendar().week).mean() # works
```

**Comment by spencerkclark** [comment, marked as answer]
The primary reason is that `cftime.datetime` objects do not implement an `isocalendar` method (see [`datetime.date.isocalendar`](https://docs.python.org/3/library/datetime.html#datetime.date.isocalendar) for more details).  You could consider raising an issue in the [`cftime` repo](https://github.com/Unidata/cftime).  If they did, it would make it pretty easy to enable in xarray.
